src/cocotb/handle.py

```diff
@@ -770,74 +944,76 @@ class ArrayObject(
-            The elements of the array appear in the list in left-to-right order.
-
-        :setter:
-            Assigns an :class:`~cocotb.types.Array`, :class:`list`, or :class:`tuple` of values to each element of the array at the end of the current delta cycle.
-            The element values are assigned in left-to-right order.
+    def get(self) -> Array[ElemValueT]:
+        """Return the current value as an :class:`~cocotb.types.Array`.
 
-        Given an HDL array ``arr``, when getting the value:
+        Given the HDL array ``arr``, getting the value is equivalent to:
 
         +--------------+---------------------+--------------------------------------------------------------------------------------------------+
-        | Verilog      | VHDL                | ``arr.value`` is equivalent to                                                                   |
+        | Verilog      | VHDL                | ``arr.get()`` is equivalent to                                                                   |
         +==============+=====================+==================================================================================================+
         | ``arr[4:7]`` | ``arr(4 to 7)``     | ``Array([arr[4].value, arr[5].value, arr[6].value, arr[7].value], range=Range(4, 'to', 7))``     |
         +--------------+---------------------+--------------------------------------------------------------------------------------------------+
         | ``arr[7:4]`` | ``arr(7 downto 4)`` | ``Array([arr[7].value, arr[6].value, arr[5].value, arr[4].value], range=Range(7, 'downto', 4))`` |
         +--------------+---------------------+--------------------------------------------------------------------------------------------------+
+        """
+        return Array((self[i].value for i in self.range), range=self.range)
+
+    def set(
+        self,
+        value: Union[
+            Union[Array[ElemValueT], Sequence[ElemValueT]],
+            Deposit[Union[Array[ElemValueT], Sequence[ElemValueT]]],
+            Force[Union[Array[ElemValueT], Sequence[ElemValueT]]],
+            Freeze,
+            Release,
+            Immediate[Union[Array[ElemValueT], Sequence[ElemValueT]]],
+        ],
+    ) -> None:
+        """Set the value using an :class:`.Array`-like value.
 
-        When setting the signal as in ``arr.value = ...``, the same index equivalence as noted in the table holds.
+        The simulation object is set, element-by-element, left-to-right, using the corresponding element of *value*.
+        The indexes of *value* and the simulation object are not taken into account, only position.
 
         .. warning::
             Assigning a value to a sub-handle:
 
-            - **Wrong**: ``dut.some_array.value[0] = 1`` (gets value as a list then updates index 0)
+            - **Wrong**: ``dut.some_array.value[0] = 1`` (gets value as an Array, updates index 0, then tosses it)
```

Comment:
```suggestion
            - **Wrong**: ``dut.some_array.value[0] = 1`` (gets value as an Array, updates index 0, then throws it away)
```
A bit more accessible english for non-native speakers